[oip] First pass at an incident response (#14)
omisego/OIP: 0005-incident_response.md

---
oip: "0005"
title: Incident Response
status: Draft
type: Operations

> author: Jake Bunce <jake@omse.co>

Reviewer: ```suggestion
author: Jake Bunce <jake@omise.co>
```

created: 2018-11-
---

# OIP-0005: Incident Response 

## Summary

Outages are common place when running services. Managing these incidents with a defined process helps to restore services faster and creates trust for the users.

## What defines an incident?

> An incident is an event that affects normal operation of an OmiseGo service. This can be a service that is used internally or externally, such as an automation service or a Plasma blockchain. Services are measured to determine normal operation through service level indicators (SLIs), from which service level objectives (SLOs) can be applied.

Reviewer: ```suggestion
An incident is an event that affects normal operation of an OmiseGO service. This can be a service that is used internally or externally, such as an automation service or a Plasma blockchain. Services are measured to determine normal operation through service level indicators (SLIs), from which service level objectives (SLOs) can be applied.
```

> For example, we may give assurances to third party partners that an endpoint API on a Watcher instance is avaiable for 99.999% of the time. And we may give an assurance that the same API is returns a response within 300ms 99% of the time. If we're not meeting those service level objectives then an incident should be raised.

Reviewer: ```suggestion
For example, we may give assurances to third party partners that an endpoint API on a Watcher instance is available for 99.999% of the time. And we may give an assurance that the same API returns a response within 300ms 99% of the time. If we're not meeting those service level objectives then an incident should be raised.
```

## Priority Levels

Priority levels are used to define the business impact to a technical service failure. Incidents can start at one level and be raised/lowered as the incident progresses.

### P0

> 

Reviewer 2: I'd suggest that executive levels are informed when P0 incidents are reported.

Author: Areas of notification added in 3452b08

> A P0 event is the most severe incident that can happen. An example of this would be a failure of multiple related services, failure of the entire Plamsa chain, or a funds beind deposited into an incorrect wallet account. These events have the capability to impact revenue steams or create significant reputational damage with users and/or partners.

Reviewer: ```suggestion
A P0 event is the most severe incident that can happen. An example of this would be a failure of multiple related services, failure of the entire Plasma chain, or funds being deposited into an incorrect wallet account. These events have the capability to impact revenue steams or create significant reputational damage with users and/or partners.
```

### P1

A P1 is an event that an entire service is unavailable. None of the features of the service are usable. 

### P2

In a P2 event a component of the service is unavailable. This may be an API endpoint, or a particular feature of the service.

### P3

A P3 is an event where a service is degraded and is not performing as expected. SLOs may be breached with API calls taking longer, creating a negative user experience. 

>  

Reviewer 2: Would it make sense to have some communications to be broadcast when incidents occur?

Reviewer: IMO, absolutely. Having some form of notification for users of the network (subscribe to receive email notifications of incidents that affect service eg), and possibly a status page people can check as well (something like status.omisego.network or whatever would be most appropriate), would be useful.

Author: Good ideas. I'll update each category to include the audience for updates.
